File: twoSum.py
class Solution:
    # n^2 version
    def twoSumSlow(self, nums: list[int], target: int) -> list[int]:
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]
    def twoSumFast(self, nums: list[int], target: int) -> list[int]:
        seen = dict()
        for i in range(len(nums)):
            seen[nums[i]] = i
        for j in range(len(nums)):
            if (target - nums[j]) in seen and seen[target - nums[j]] != j:
                return [j, seen[target - nums[j]]]

File: twoSum.cpp
#include<utility>
# include<vector>

class Solution {
    // Doesn't work yet. Currently always reaches return garbage
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        std::vector<std::pair<int, int>> seen = std::vector<std::pair<int, int>>();
        for (int i = 0; i < int(nums.size()); i++) {
            for (int j = 0; j < int(seen.size()); j++) {
                if (nums[i] - seen[j].first == target) {
                    std::vector<int> ans = {i, seen[j].second};
                    return ans;
                }
                seen.emplace_back(std::pair<int, int>({nums[i], i}));
            }
        }
        std::vector<int> garbage = {-1, -1}; // return garbage answer if no real answer found
        return garbage;
    }
};